test/test_dataloader.py

```diff
@@ -220,6 +232,18 @@ def _test_timeout():
         finally:
             p.terminate()
 
+    def test_worker_init_fn(self):
+        # test custom init function
+        def init_fn(worker_id):
+            torch.manual_seed(12345)
+
+        dataset = SeedDataset(4)
+        dataloader = DataLoader(dataset, batch_size=2, num_workers=2,
+                                worker_init_fn=init_fn)
+        for batch in dataloader:
+            self.assertEqual(12345, batch[0])
+            self.assertEqual(12345, batch[1])
```

Comment:
Can you add a test for the case when `init_fn` is not specified, so we ensure that workers are seeded differently?

Reply:
Is there a way to make sure that each worker process data? I thought about that but it will fail if all data is given by the same worker..

Reply:
Ohh I see, that makes sense. One way would be to bake some multiprocess synchronization primitives into the dataset object (e.g. a barrier).

Reply:
I have it working now. However, there is one more question... what if main process generates duplicate seeds for the workers? :D

Reply:
Resolved. See below for details.